src/hist/intervals.py

```diff
@@ -55,11 +55,12 @@ def poisson_interval(
     # https://github.com/CoffeaTeam/coffea/blob/8c58807e199a7694bf15e3803dbaf706d34bbfa0/LICENSE
     if coverage is None:
         coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
-    scale = np.empty_like(values)
-    scale[values != 0] = variances[values != 0] / values[values != 0]
+    scale = np.ones_like(values)
+    mask = np.isfinite(values) & (values != 0)
+    scale[mask] = variances[mask] / values[mask]
     if np.sum(values == 0) > 0:
```

Comment:
After revisiting this, I'm somewhat dubious if we want to keep the "magic" behavior implemented here. If not for this then the functions would be elementwise (i.e. ufunc-like), which I think makes a bit more sense.

Reply:
I'm generally okay with removing magic. :) @matthewfeickert ?

Reply:
@nsmith- You're saying to just remove this whole logic block and say that if you have values that are zero then you'll need to figure that out? If so, I think that's fine as it is probably better to be explicit with the users about their data/inputs being their responsibility.

Reply:
Yeah, specifically this whole indented block starting at L60. I think in the case where a value is zero we should just return a fixed value. Sometimes people propose `1.84` (upper interval assuming unit scale) but I think since we don't have a known scale the best is something like `0` or `nan`. Just another example why frequentist confidence levels are a bit silly.

Reply:
SGTM. Also yes. :)

Reply:
I made `variances` an optional argument so people can still use `poisson_interval` for unweighted counts and always get an upper limit. Only when variances is supplied do we have this ambiguity.